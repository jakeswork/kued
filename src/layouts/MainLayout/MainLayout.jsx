import React, { Fragment } from 'react';

import { propTypes, defaultProps } from './types';
import SideMenu from './components/SideMenu';

const MainLayout = ({ classes, children }) => (
  <Fragment>
    <SideMenu />
    <main className={classes.main}>
      <div className={classes.root}>
        { children }
      </div>
    </main>
  </Fragment>
);

MainLayout.propTypes = propTypes;

MainLayout.defaultProps = defaultProps;

export default MainLayout;
