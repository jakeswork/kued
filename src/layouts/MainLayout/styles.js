export default theme => ({
  main: {
    marginLeft: 300,
    padding: '32px 16px',
  },
  [theme.media.mobile]: {
    main: {
      marginLeft: 0,
    },
  },
});
