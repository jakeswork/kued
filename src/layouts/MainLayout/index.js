import withStyles from 'react-jss';

import MainLayout from './MainLayout';
import styles from './styles';

export default withStyles(styles)(MainLayout);
