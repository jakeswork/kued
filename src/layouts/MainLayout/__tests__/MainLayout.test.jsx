import React from 'react';
import { shallow } from 'enzyme';

import MainLayout from '../MainLayout';

const defaultProps = {
  classes: {},
};

describe('The MainLayout instance', () => {
  describe('when called with default properties', () => {
    it('should render without throwing any errors', () => {
      const wrapper = shallow(<MainLayout {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
