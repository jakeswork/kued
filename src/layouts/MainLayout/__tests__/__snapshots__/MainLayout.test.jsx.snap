// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MainLayout instance when called with default properties should render without throwing any errors 1`] = `
<Fragment>
  <Jss(SideMenu) />
  <main />
</Fragment>
`;
